File: android/src/main/kotlin/com/kurban/xue_hua_video_snapshot/PigeonCodecHelpers.kt
package com.kurban.xue_hua_video_snapshot

/** Coerces Pigeon message arguments to [Long]. */
internal fun pigeonLong(value: Any?): Long = when (value) {
    is Long -> value
    is Int -> value.toLong()
    is Number -> value.toLong()
    else -> throw IllegalArgumentException("Expected numeric pigeon argument, got $value")
}

File: lib/shared/apple/PigeonCodecHelpers.swift
import Foundation

/// Coerces Pigeon message arguments to [Int64].
///
/// Dart [_PigeonCodec] always writes `int` as StandardMessageCodec type-4 Int64,
/// but Apple codecs may deliver [NSNumber] or [Int].
enum PigeonCodecHelpers {
    static func int64(from value: Any?) -> Int64 {
        switch value {
        case let v as Int64:
            return v
        case let v as Int:
            return Int64(v)
        case let n as NSNumber:
            return n.int64Value
        default:
            fatalError("Expected numeric pigeon argument, got \(String(describing: value))")
        }
    }
}
